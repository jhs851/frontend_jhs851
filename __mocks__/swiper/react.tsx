import React from 'react'

const mock: React.FC = ({ children }) => <div>{children}</div>

export default mock
