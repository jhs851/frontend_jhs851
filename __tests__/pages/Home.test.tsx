import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import Home from 'pages'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('Home 페이지', () => {
  it('렌더링이 올바르게 된다', () => {
    expect(render(<Home />).container).toMatchSnapshot()
  })

  it('채팅 글씨가 있는 버튼을 누르면 list 페이지로 라우팅된다', () => {
    const push = jest.fn()
    useRouter.mockImplementationOnce(() => ({
      route: '/',
      pathname: '/',
      query: '',
      asPath: '',
      push,
    }))

    const { getByText } = render(<Home />)

    const button = getByText('채팅')
    fireEvent.click(button)
    expect(push).toHaveBeenCalledWith('/list', '/list', expect.anything())
  })
})
