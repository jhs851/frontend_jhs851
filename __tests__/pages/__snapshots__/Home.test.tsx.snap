// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 페이지 렌더링이 올바르게 된다 1`] = `
.c1 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 40px;
  font-weight: 700;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
}

.c2 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 22px;
  font-weight: 500;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
}

.c3 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 20px;
  font-weight: 500;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
}

.c5 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 14px;
  font-weight: 500;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
}

.c0 {
  height: 100vh;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  width: 110px;
  padding: 8px;
  border-radius: 20px;
  margin-top: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
      size="40"
    >
      안녕하세요!
    </p>
    <p
      class="c2"
      size="22"
    >
      프론트엔드 개발자 
      <b>
        정형석
      </b>
      입니다!
    </p>
    <p
      class="c3"
      size="20"
    >
      잘 부탁드립니다 😊😊😊
    </p>
    <a
      class="c4"
      href="/list"
    >
      <p
        class="c5"
        color="white"
        size="14"
      >
        채팅
      </p>
    </a>
  </div>
</div>
`;
